Feature/intake

Henyahyah opened the pull request:
Before issuing a pull request, please see the contributing page.

Henyahyah committed:
More Quantum Bugs Fixed

We just fixed the importing issue with more the the code and prevented a lot of errors that we wouldn't've seen coming.

Henyahyah committed:
Merge branch 'feature/intake' of https://github.com/RedArrowRobotics/2024-21353-RobotCode into feature/intake

Henyahyah committed:
Testing Fixes

Fixed some bugs discovered from testing and fixed the configuration.

RedArrowRobotics commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Actively_Active_Intake.java:
```diff
@@ -8,44 +8,46 @@
-    void rotateIn(){
+    void rotateIn(Telemetry telemetry){
         rotate.setPosition(90);
         telemetry.addData("Intake Position", "%4.2f", rotate.getPosition());
     }
-    void startSpin(){
+    void startSpin(Telemetry telemetry){
         spin.setPower(1);
         telemetry.addData("Intake Power", "%4.2f", spin.getPower());
     }
-    void stopSpin(){
+    void stopSpin(Telemetry telemetry){
         spin.setPower(0);
         telemetry.addData("Intake Power", "%4.2f", spin.getPower());
     }
```
Something to think about - we'll need to be able to spin in the opposite direction as well to 'unload' the sample.

RedArrowRobotics commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ViperArrrrrm.java:
```diff
@@ -19,17 +21,16 @@ public class ViperArrrrrm extends LinearOpMode {
         viperArm = hwm.get(DcMotor.class, Constants.VIPER_ARM);
-        viperArm.setDirection(DcMotor.Direction.FORWARD);
+        viperArm.setDirection(DcMotor.Direction.REVERSE);
     }
-    void operateArm(double armPower){
+    void operateArm(Telemetry telemetry, double armPower){
         double max = 1;
         max = Math.max(max, Math.abs(armPower));
         if (armPower > max) {
             armPower = max;
         }
         viperArm.setPower(armPower);
         telemetry.addData("arm power", "%4.2f", armPower);
-        telemetry.update();
     }
```
We have an encoder on this motor, as well as a limit switch on the bottom to zero out the encoder. 
Examples:
* Touch Sensor: https://github.com/RedArrowRobotics/2024-21353-RobotCode/blob/master/FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/external/samples/SensorTouch.java
* Encoder (The example is using drive motors, so there will be some differences in how we want to control it, but this should get you started): https://github.com/RedArrowRobotics/2024-21353-RobotCode/blob/master/FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/external/samples/RobotAutoDriveByEncoder_Linear.java

RedArrowRobotics replied:
essentially - we will want to zero out the encoder whenever the button is pressed, then add some telemetry so we can figure out what encoder counts correspond to heights of the two buckets. Then use 'runToPosition' to go directly to either the upper or lower basket.

Henyahyah committed:
Maggie Disaster Imminent

She yelled at us to save so here's a mini save. I'll give real desc. in final save for this session..

Henyahyah committed:
IT WORKSS

We can now score samples into buckets, so everything works. We just need to add the encoder limiter for the Viper Arm then autonomous.

Henyahyah committed:
Fixed more bugs and smaller details like fixing the controller layout

i really simplified the controls

Henyahyah committed:
Notes

I just added notes for what to fix after testing today and removed some ghost code before I optimized the controls.
